# HG changeset patch
# User Petr Sumbera <petr.sumbera@oracle.com>
# Date 1604331515 -3600
#      Mon Nov 02 16:38:35 2020 +0100
# Node ID df0faa3e0349d95e433ef9924bfa83aa7becd845
# Parent  936f52771552e8daa450888af1b6cc6564d6b3f5
RESERVE_SEGMENT for SPARC - limit stack below 47bits of address space

--- firefox-140.1.0/browser/app/moz.build
+++ firefox-140.1.0/browser/app/moz.build
@@ -151,6 +151,10 @@ # Set it to 256k.  See bug 127069.
 if CONFIG["OS_ARCH"] == "WINNT" and CONFIG["CC_TYPE"] not in ("clang", "gcc"):
     LDFLAGS += ["/HEAP:0x40000"]
 
+# Enforce 47bits address space to resulting binary (Bug 577056).
+if CONFIG['OS_ARCH'] == 'SunOS' and CONFIG['TARGET_CPU'] == 'sparc64':
+    LDFLAGS += ['-Wl,-M/usr/lib/ld/map.vas47']
+
 DisableStlWrapping()
 
 if CONFIG["HAVE_CLOCK_MONOTONIC"]:
--- firefox-140.1.0/js/src/shell/moz.build
+++ firefox-140.1.0/js/src/shell/moz.build
@@ -64,3 +64,7 @@ if CONFIG["OS_TARGET"] == "Android":
     LDFLAGS += [
         "-Wl,-rpath=$ORIGIN",
     ]
+
+# Enforce 47bits address space to resulting binary (Bug 577056).
+if CONFIG['OS_ARCH'] == 'SunOS' and CONFIG['TARGET_CPU'] == 'sparc64':
+    LDFLAGS += ['-Wl,-M/usr/lib/ld/map.vas47']
--- firefox-140.1.0/js/xpconnect/shell/moz.build
+++ firefox-140.1.0/js/xpconnect/shell/moz.build
@@ -32,6 +32,10 @@ # Always enter a Windows program through
     # a console application.
     WIN32_EXE_LDFLAGS += ["-ENTRY:wmainCRTStartup"]
 
+# Enforce 47bits address space to resulting binary (Bug 577056).
+if CONFIG['OS_ARCH'] == 'SunOS' and CONFIG['TARGET_CPU'] == 'sparc64':
+    LDFLAGS += ['-Wl,-M/usr/lib/ld/map.vas47']
+
 # DELAYLOAD_DLLS in this block ensure that the DLL blocklist initializes
 if CONFIG["OS_ARCH"] == "WINNT":
     if CONFIG["MOZ_SANDBOX"]:
